source/dsl/GitHubDSL.ts

```diff
@@ -1,9 +1,27 @@
+import { GitCommit } from "./Commit"
+
+/** A GitHub specific implmentation of a git commit */
+export interface GitHubCommit {
+  /** The raw commit metadata */
+  commit: GitCommit,
+  /** The SHA for the commit */
+  sha: string,
+  /** the url for the commit on GitHub */
+  url: string,
+  /** The GitHub user who wrote the code */
+  author: GitHubUser,
+  /** The GitHub user who shipped the code */
+  committer: GitHubUser,
+  /** An array of parent commit shas */
+  parents: Array<any>
```

Comment:
Two things:

- Would this be an `Array<string>`?
- I never know when to use `Array<string>` vs. `string[]` 🤔